diff -Naur a/clean_exit.c b/clean_exit.c
--- a/clean_exit.c	2018-02-13 00:19:36.000000000 +1100
+++ b/clean_exit.c	2018-02-13 00:20:06.000000000 +1100
@@ -15,7 +15,8 @@
 #include <stdio.h>
 #include <unistd.h>
 
-extern void exit();
+#include <unistd.h>
+#include <stdlib.h>
 
 #include "tcpd.h"
 
diff -Naur a/fix_options.c b/fix_options.c
--- a/fix_options.c	2018-02-13 00:19:36.000000000 +1100
+++ b/fix_options.c	2018-02-13 00:23:00.000000000 +1100
@@ -38,12 +38,8 @@
 #ifdef IP_OPTIONS
     unsigned char optbuf[BUFFER_SIZE / 3], *cp;
     char    lbuf[BUFFER_SIZE], *lp;
-#ifdef __GLIBC__
     socklen_t optsize = sizeof(optbuf);
     int     ipproto;
-#else
-    int     optsize = sizeof(optbuf), ipproto;
-#endif
     struct protoent *ip;
     int     fd = request->fd;
     unsigned int opt;
diff -Naur a/hosts_access.c b/hosts_access.c
--- a/hosts_access.c	2018-02-13 00:19:36.000000000 +1100
+++ b/hosts_access.c	2018-02-13 00:20:06.000000000 +1100
@@ -23,6 +23,7 @@
 
 /* System libraries. */
 
+#include <stdlib.h>
 #include <sys/types.h>
 #ifdef INT32_T
     typedef uint32_t u_int32_t;
@@ -44,8 +45,7 @@
 #include <netdb.h>
 #endif
 
-extern char *fgets();
-extern int errno;
+static int match_pattern_ylo(const char *s, const char *pattern);
 
 #ifndef	INADDR_NONE
 #define	INADDR_NONE	(-1)		/* XXX should be 0xffffffff */
diff -Naur a/inetcf.c b/inetcf.c
--- a/inetcf.c	2018-02-13 00:19:36.000000000 +1100
+++ b/inetcf.c	2018-02-13 00:20:06.000000000 +1100
@@ -9,15 +9,14 @@
 static char sccsid[] = "@(#) inetcf.c 1.7 97/02/12 02:13:23";
 #endif
 
+#include <stdlib.h>
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <stdio.h>
 #include <errno.h>
 #include <string.h>
 
-extern int errno;
-extern void exit();
-
+#include "scaffold.h"
 #include "tcpd.h"
 #include "inetcf.h"
 #include "scaffold.h"
diff -Naur a/misc.c b/misc.c
--- a/misc.c	2018-02-13 00:19:36.000000000 +1100
+++ b/misc.c	2018-02-13 00:20:06.000000000 +1100
@@ -15,11 +15,10 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
+#include <stdlib.h>
 
 #include "tcpd.h"
 
-extern char *fgets();
-
 #ifndef	INADDR_NONE
 #define	INADDR_NONE	(-1)		/* XXX should be 0xffffffff */
 #endif
diff -Naur a/options.c b/options.c
--- a/options.c	2018-02-13 00:19:36.000000000 +1100
+++ b/options.c	2018-02-13 00:20:06.000000000 +1100
@@ -34,6 +34,8 @@
 
 /* System libraries. */
 
+#include <unistd.h>
+#include <stdlib.h>
 #include <sys/types.h>
 #include <sys/param.h>
 #include <sys/socket.h>
diff -Naur a/percent_x.c b/percent_x.c
--- a/percent_x.c	2018-02-13 00:19:36.000000000 +1100
+++ b/percent_x.c	2018-02-13 00:24:34.000000000 +1100
@@ -16,13 +16,13 @@
 
 /* System libraries. */
 
+#include <unistd.h>
+#include <stdlib.h>
 #include <stdio.h>
 #include <unistd.h>
 #include <syslog.h>
 #include <string.h>
 
-extern void exit();
-
 /* Local stuff. */
 
 #include "tcpd.h"
diff -Naur a/rfc931.c b/rfc931.c
--- a/rfc931.c	2018-02-13 00:19:36.000000000 +1100
+++ b/rfc931.c	2018-02-13 00:20:06.000000000 +1100
@@ -15,6 +15,7 @@
 
 /* System libraries. */
 
+#include <unistd.h>
 #include <stdio.h>
 #include <unistd.h>
 #include <syslog.h>
diff -Naur a/safe_finger.c b/safe_finger.c
--- a/safe_finger.c	2018-02-13 00:19:36.000000000 +1100
+++ b/safe_finger.c	2018-02-13 00:20:06.000000000 +1100
@@ -20,6 +20,11 @@
 
 /* System libraries */
 
+#include <unistd.h>
+#include <fcntl.h>
+#include <stdlib.h>
+#include <sys/wait.h>
+#include <grp.h>
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <sys/wait.h>
@@ -33,7 +38,7 @@
 #include <grp.h>
 #include <syslog.h>
 
-extern void exit();
+int pipe_stdin(char **argv);
 
 /* Local stuff */
 
diff -Naur a/scaffold.c b/scaffold.c
--- a/scaffold.c	2018-02-13 00:19:36.000000000 +1100
+++ b/scaffold.c	2018-02-13 00:20:06.000000000 +1100
@@ -10,6 +10,7 @@
 
 /* System libraries. */
 
+#include <stdlib.h>
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <sys/socket.h>
diff -Naur a/shell_cmd.c b/shell_cmd.c
--- a/shell_cmd.c	2018-02-13 00:19:36.000000000 +1100
+++ b/shell_cmd.c	2018-02-13 00:20:06.000000000 +1100
@@ -14,6 +14,10 @@
 
 /* System libraries. */
 
+#include <unistd.h>
+#include <stdlib.h>
+#include <fcntl.h>
+#include <sys/wait.h>
 #include <sys/types.h>
 #include <sys/param.h>
 #include <signal.h>
@@ -26,8 +30,6 @@
 #include <sys/stat.h>
 #include <fcntl.h>
 
-extern void exit();
-
 /* Local stuff. */
 
 #include "tcpd.h"
diff -Naur a/socket.c b/socket.c
--- a/socket.c	2018-02-13 00:19:36.000000000 +1100
+++ b/socket.c	2018-02-13 00:27:27.000000000 +1100
@@ -88,18 +88,9 @@
 void    sock_host(request)
 struct request_info *request;
 {
-#ifdef INET6
-    static struct sockaddr_storage client;
-    static struct sockaddr_storage server;
-#else
     static struct sockaddr_in client;
     static struct sockaddr_in server;
-#endif
-#ifdef __GLIBC__
-    size_t  len;
-#else
-    int     len;
-#endif
+    socklen_t  len;
     char    buf[BUFSIZ];
     int     fd = request->fd;
 
@@ -425,16 +416,8 @@
 int     fd;
 {
     char    buf[BUFSIZ];
-#ifdef INET6
-    struct sockaddr_storage sin;
-#else
     struct sockaddr_in sin;
-#endif
-#ifdef __GLIBC__
-    size_t  size = sizeof(sin);
-#else
-    int     size = sizeof(sin);
-#endif
+    socklen_t  size = sizeof(sin);
 
     /*
      * Eat up the not-yet received datagram. Some systems insist on a
diff -Naur a/tcpd.c b/tcpd.c
--- a/tcpd.c	2018-02-13 00:19:36.000000000 +1100
+++ b/tcpd.c	2018-02-13 00:20:06.000000000 +1100
@@ -16,6 +16,7 @@
 
 /* System libraries. */
 
+#include <unistd.h>
 #include <sys/types.h>
 #include <sys/param.h>
 #include <sys/stat.h>
@@ -41,6 +42,8 @@
 
 void fix_options(struct request_info *);
 
+void fix_options(struct request_info *request);
+
 int     allow_severity = SEVERITY;	/* run-time adjustable */
 int     deny_severity = LOG_WARNING;	/* ditto */
 
diff -Naur a/tcpdchk.c b/tcpdchk.c
--- a/tcpdchk.c	2018-02-13 00:19:36.000000000 +1100
+++ b/tcpdchk.c	2018-02-13 00:20:06.000000000 +1100
@@ -20,6 +20,8 @@
 
 /* System libraries. */
 
+#include <unistd.h>
+#include <stdlib.h>
 #include <sys/types.h>
 #include <sys/stat.h>
 #ifdef INET6
@@ -36,10 +38,6 @@
 #include <netdb.h>
 #include <string.h>
 
-extern int errno;
-extern void exit();
-extern int optind;
-extern char *optarg;
 
 #ifndef INADDR_NONE
 #define INADDR_NONE     (-1)		/* XXX should be 0xffffffff */
diff -Naur a/tcpdmatch.c b/tcpdmatch.c
--- a/tcpdmatch.c	2018-02-13 00:19:36.000000000 +1100
+++ b/tcpdmatch.c	2018-02-13 00:20:06.000000000 +1100
@@ -19,6 +19,8 @@
 
 /* System libraries. */
 
+#include <unistd.h>
+#include <stdlib.h>
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <sys/socket.h>
@@ -31,10 +33,6 @@
 #include <setjmp.h>
 #include <string.h>
 
-extern void exit();
-extern int optind;
-extern char *optarg;
-
 #ifndef	INADDR_NONE
 #define	INADDR_NONE	(-1)		/* XXX should be 0xffffffff */
 #endif
diff -Naur a/update.c b/update.c
--- a/update.c	2018-02-13 00:19:36.000000000 +1100
+++ b/update.c	2018-02-13 00:20:06.000000000 +1100
@@ -19,6 +19,7 @@
 
 /* System libraries */
 
+#include <unistd.h>
 #include <stdio.h>
 #include <unistd.h>
 #include <syslog.h>

