diff -Naur a/source/blender/gpu/GPU_vertex_buffer.h b/source/blender/gpu/GPU_vertex_buffer.h
--- a/source/blender/gpu/GPU_vertex_buffer.h	2020-11-26 14:23:06.051085618 +1100
+++ b/source/blender/gpu/GPU_vertex_buffer.h	2020-11-26 14:23:40.398931106 +1100
@@ -108,10 +108,8 @@
   uint stride;
   unsigned char *data;
   unsigned char *data_init;
-#ifdef DEBUG
   /* Only for overflow check */
   unsigned char *_data_end;
-#endif
 } GPUVertBufRaw;
 
 GPU_INLINE void *GPU_vertbuf_raw_step(GPUVertBufRaw *a)
