diff -pur festival-orig/src/modules/MultiSyn/base_class/EST_TSimpleVector.cc festival/src/modules/MultiSyn/base_class/EST_TSimpleVector.cc
--- festival-orig/src/modules/MultiSyn/base_class/EST_TSimpleVector.cc	2017-09-04 17:54:03.000000000 +0200
+++ festival/src/modules/MultiSyn/base_class/EST_TSimpleVector.cc	2018-03-14 09:31:57.950640675 +0100
@@ -42,6 +42,7 @@
 
 #include "EST_TSimpleVector.h"
 #include "EST_matrix_support.h"
+#include <cstring>
 #include <fstream>
 #include "EST_cutils.h"
 #include <string.h>
@@ -50,7 +51,7 @@ template<class T> void EST_TSimpleVector
 {
   if (this->p_column_step==1 && a.p_column_step==1)
     {
-    resize(a.n(), FALSE);
+    this->resize(a.n(), FALSE);
     memcpy((void *)(this->p_memory), (const void *)(a.p_memory), this->n() * sizeof(T));
     }
 else

