diff -Naur a/src/cmake/modules/FindPybind11.cmake b/src/cmake/modules/FindPybind11.cmake
--- a/src/cmake/modules/FindPybind11.cmake	2020-01-19 13:24:32.218308164 +0800
+++ b/src/cmake/modules/FindPybind11.cmake	2020-01-19 13:25:28.824571975 +0800
@@ -7,7 +7,13 @@
 # PYBIND11_VERSION - version found
 
 find_path (PYBIND11_INCLUDE_DIR pybind11/pybind11.h
-           HINTS "${PROJECT_SOURCE_DIR}/ext/pybind11/include"
+           PATHS "/usr/include/python3.8m"
+           )
+find_path (PYBIND11_INCLUDE_DIR pybind11/pybind11.h
+           PATHS "/usr/include/python3.7m"
+           )
+find_path (PYBIND11_INCLUDE_DIR pybind11/pybind11.h
+           PATHS "/usr/include/python3.6m"
            )
 
 if (PYBIND11_INCLUDE_DIR)
