diff -Naur a/src/doc/Makefile b/src/doc/Makefile
--- a/src/doc/Makefile	2020-02-03 09:57:21.869224398 +0800
+++ b/src/doc/Makefile	2020-02-03 09:58:23.315340026 +0800
@@ -3,10 +3,10 @@
 include ../make/detectplatform.mk
 BUILDDIR := ../../build/${platform}
 ifeq (${OIIOTOOL},)
-  ifneq (${shell ls ${BUILDDIR}/src/oiiotool/oiiotool},)
-    OIIOTOOL := "${BUILDDIR}/src/oiiotool/oiiotool"
+  ifneq (${shell ls ${BUILDDIR}/bin/oiiotool},)
+    OIIOTOOL := "${BUILDDIR}/bin/oiiotool"
   else
-    OIIOTOOL := "../../build/src/oiiotool/oiiotool"
+    OIIOTOOL := "../../build/bin/oiiotool"
   endif
 endif
 
