diff -pur speech_tools-orig/base_class/EST_TSimpleMatrix.cc speech_tools/base_class/EST_TSimpleMatrix.cc
--- speech_tools-orig/base_class/EST_TSimpleMatrix.cc	2013-04-13 14:51:56.000000000 +0200
+++ speech_tools/base_class/EST_TSimpleMatrix.cc	2018-03-14 07:48:57.169248488 +0100
@@ -42,6 +42,7 @@
 
 #include "EST_TSimpleMatrix.h"
 #include "EST_TVector.h"
+#include <cstring>
 #include <fstream>
 #include <iostream>
 #include "EST_cutils.h"
diff -pur speech_tools-orig/base_class/EST_TSimpleVector.cc speech_tools/base_class/EST_TSimpleVector.cc
--- speech_tools-orig/base_class/EST_TSimpleVector.cc	2013-04-13 14:52:21.000000000 +0200
+++ speech_tools/base_class/EST_TSimpleVector.cc	2018-03-14 07:48:57.169248488 +0100
@@ -42,6 +42,7 @@
 
 #include "EST_TSimpleVector.h"
 #include "EST_matrix_support.h"
+#include <cstring>
 #include <fstream>
 #include "EST_cutils.h"
 #include <string.h>
Only in speech_tools/base_class: EST_TSimpleVector.cc.orig
