diff -pur speech_tools-orig/main/siod_main.cc speech_tools/main/siod_main.cc
--- speech_tools-orig/main/siod_main.cc	2013-10-15 00:13:51.000000000 +0200
+++ speech_tools/main/siod_main.cc	2018-03-14 07:55:28.652641361 +0100
@@ -222,10 +222,10 @@ static void siod_lisp_vars(void)
 			    cons(flocons(subminor),NIL))));
 
     EST_Pathname etcdircommon = est_libdir;
-    etcdircommon += "etc";
+    etcdircommon += "etc/";
 
-    EST_Pathname etcdir = etcdircommon;
-    etcdir += est_ostype;
+    EST_Pathname etcdir = {{HORRIBLELIBARCHKLUDGE}};
+    etcdir += "festival/etc/";
     
     //  Modify my PATH to include these directories
     siod_set_lval("etc-path",cons(rintern(etcdir),
Only in speech_tools/main: siod_main.cc.orig
