diff --git speech_tools-orig/config/systems/GentooLinux.mak speech_tools/config/systems/GentooLinux.mak
--- /dev/null
+++ speech_tools/config/systems/GentooLinux.mak
@@ -0,0 +1,42 @@
+ ###########################################################################
+ ##                                                                       ##
+ ##                Centre for Speech Technology Research                  ##
+ ##                     University of Edinburgh, UK                       ##
+ ##                         Copyright (c) 1996                            ##
+ ##                        All Rights Reserved.                           ##
+ ##                                                                       ##
+ ##  Permission is hereby granted, free of charge, to use and distribute  ##
+ ##  this software and its documentation without restriction, including   ##
+ ##  without limitation the rights to use, copy, modify, merge, publish,  ##
+ ##  distribute, sublicense, and/or sell copies of this work, and to      ##
+ ##  permit persons to whom this work is furnished to do so, subject to   ##
+ ##  the following conditions:                                            ##
+ ##   1. The code must retain the above copyright notice, this list of    ##
+ ##      conditions and the following disclaimer.                         ##
+ ##   2. Any modifications must be clearly marked as such.                ##
+ ##   3. Original authors' names are not deleted.                         ##
+ ##   4. The authors' names are not used to endorse or promote products   ##
+ ##      derived from this software without specific prior written        ##
+ ##      permission.                                                      ##
+ ##                                                                       ##
+ ##  THE UNIVERSITY OF EDINBURGH AND THE CONTRIBUTORS TO THIS WORK        ##
+ ##  DISCLAIM ALL WARRANTIES WITH REGARD TO THIS SOFTWARE, INCLUDING      ##
+ ##  ALL IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS, IN NO EVENT   ##
+ ##  SHALL THE UNIVERSITY OF EDINBURGH NOR THE CONTRIBUTORS BE LIABLE     ##
+ ##  FOR ANY SPECIAL, INDIRECT OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES    ##
+ ##  WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN   ##
+ ##  AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION,          ##
+ ##  ARISING OUT OF OR IN CONNECTION WITH THE USE OR PERFORMANCE OF       ##
+ ##  THIS SOFTWARE.                                                       ##
+ ##                                                                       ##
+ ###########################################################################
+ ##                                                                       ##
+ ##                 Author: William Hubbs <williamh@gentoo.org>           ##
+ ## --------------------------------------------------------------------  ##
+ ## Settings for Gentoo Linux distributions.                              ##
+ ##                                                                       ##
+ ###########################################################################
+
+MANDIR=$(INSTALL_PREFIX)/share/man
+# Gentoo does not use termcap
+OS_LIBS = $(LDFLAGS) -ldl -lncurses
diff -pur speech_tools-orig/config/systems/Makefile speech_tools/config/systems/Makefile
--- speech_tools-orig/config/systems/Makefile	2010-04-19 21:18:26.000000000 +0200
+++ speech_tools/config/systems/Makefile	2018-03-14 07:42:46.502929681 +0100
@@ -41,6 +41,7 @@ SYSTEMS =					\
 	Linux.mak				\
 	RedHatLinux.mak				\
 	DebianGNULinux.mak			\
+	GentooLinux.mak			\
 	alpha_Linux.mak				\
 	alpha_OSF1V4.0.mak			\
 	alpha_RedHatLinux.mak			\
@@ -96,6 +97,7 @@ SYSTEMS =					\
 	sparc_SunOS5.mak			\
 	sparc_unknown.mak			\
 	unknown_DebianGNULinux.mak		\
+	unknown_GentooLinux.mak		\
 	unknown_Linux.mak			\
 	unknown_RedHatLinux.mak			\
         power_macintosh_Darwin.mak              \
diff -pur speech_tools-orig/config/systems/unknown_GentooLinux.mak speech_tools/config/systems/unknown_GentooLinux.mak
--- /dev/null
+++ speech_tools/config/systems/unknown_GentooLinux.mak
@@ -0,0 +1,41 @@
+ ###########################################################################
+ ##                                                                       ##
+ ##                Centre for Speech Technology Research                  ##
+ ##                     University of Edinburgh, UK                       ##
+ ##                         Copyright (c) 1996                            ##
+ ##                        All Rights Reserved.                           ##
+ ##                                                                       ##
+ ##  Permission is hereby granted, free of charge, to use and distribute  ##
+ ##  this software and its documentation without restriction, including   ##
+ ##  without limitation the rights to use, copy, modify, merge, publish,  ##
+ ##  distribute, sublicense, and/or sell copies of this work, and to      ##
+ ##  permit persons to whom this work is furnished to do so, subject to   ##
+ ##  the following conditions:                                            ##
+ ##   1. The code must retain the above copyright notice, this list of    ##
+ ##      conditions and the following disclaimer.                         ##
+ ##   2. Any modifications must be clearly marked as such.                ##
+ ##   3. Original authors' names are not deleted.                         ##
+ ##   4. The authors' names are not used to endorse or promote products   ##
+ ##      derived from this software without specific prior written        ##
+ ##      permission.                                                      ##
+ ##                                                                       ##
+ ##  THE UNIVERSITY OF EDINBURGH AND THE CONTRIBUTORS TO THIS WORK        ##
+ ##  DISCLAIM ALL WARRANTIES WITH REGARD TO THIS SOFTWARE, INCLUDING      ##
+ ##  ALL IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS, IN NO EVENT   ##
+ ##  SHALL THE UNIVERSITY OF EDINBURGH NOR THE CONTRIBUTORS BE LIABLE     ##
+ ##  FOR ANY SPECIAL, INDIRECT OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES    ##
+ ##  WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN   ##
+ ##  AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION,          ##
+ ##  ARISING OUT OF OR IN CONNECTION WITH THE USE OR PERFORMANCE OF       ##
+ ##  THIS SOFTWARE.                                                       ##
+ ##                                                                       ##
+ ###########################################################################
+ ##                                                                       ##
+ ##                 Author: William Hubbs <williamh@gentoo.org>           ##
+ ## --------------------------------------------------------------------  ##
+ ## Settings for Gentoo Linux distributions.                              ##
+ ##                                                                       ##
+ ###########################################################################
+
+ include $(EST)/config/systems/Linux.mak
+ include $(EST)/config/systems/GentooLinux.mak
diff -pur speech_tools-orig/config/system.sh speech_tools/config/system.sh
--- speech_tools-orig/config/system.sh	2002-12-27 22:20:14.000000000 +0100
+++ speech_tools/config/system.sh	2018-03-14 07:42:46.502929681 +0100
@@ -70,6 +70,10 @@ if [ "$OSTYPE" = Linux ]
 	then
 	OSTYPE=DebianGNULinux
 	OSREV=`cat /etc/debian_version`
+    elif [ -f "/etc/gentoo-release" ]
+    then
+    OSTYPE=GentooLinux
+    OSREV=
     else
 	# Generic unknown GNU/Linux system.
 	OSTYPE=Linux
