diff -Naur a/client/notifier.py b/client/notifier.py
--- a/client/notifier.py	2020-04-20 11:00:00.871650262 +1000
+++ b/client/notifier.py	2020-04-20 11:02:22.602039377 +1000
@@ -1,7 +1,7 @@
 # -*- coding: utf-8-*-
-import Queue
+import queue
 import atexit
-from modules import Gmail
+from client.modules import Gmail
 from apscheduler.schedulers.background import BackgroundScheduler
 import logging
 
