diff -Naur a/client/mic.py b/client/mic.py
--- a/client/mic.py	2020-04-19 16:35:40.094664190 +1000
+++ b/client/mic.py	2020-04-19 16:35:57.273592833 +1000
@@ -148,7 +148,7 @@
 
         # no use continuing if no flag raised
         if not didDetect:
-            print "No disturbance detected"
+            print("No disturbance detected")
             stream.stop_stream()
             stream.close()
             return (None, None)
