diff -Naur a/client/stt.py b/client/stt.py
--- a/client/stt.py	2020-04-19 17:41:31.049609040 +1000
+++ b/client/stt.py	2020-04-19 17:41:18.020413318 +1000
@@ -6,7 +6,7 @@
 import tempfile
 import logging
 import urllib
-import urlparse
+from urllib.parse import urlparse
 import re
 import subprocess
 from abc import ABCMeta, abstractmethod
diff -Naur a/client/tts.py b/client/tts.py
--- a/client/tts.py	2020-04-19 17:41:34.479397315 +1000
+++ b/client/tts.py	2020-04-19 17:41:18.020413318 +1000
@@ -16,7 +16,7 @@
 import logging
 import wave
 import urllib
-import urlparse
+from urllib.parse import urlparse
 import requests
 from abc import ABCMeta, abstractmethod
 
