diff -Naur a/client/tts.py b/client/tts.py
--- a/client/tts.py	2020-04-19 16:55:12.441904408 +1000
+++ b/client/tts.py	2020-04-19 16:55:27.220989371 +1000
@@ -100,7 +100,7 @@
             wav.setframerate(mf.samplerate())
             wav.setnchannels(1 if mf.mode() == mad.MODE_SINGLE_CHANNEL else 2)
             # 4L is the sample width of 32 bit audio
-            wav.setsampwidth(4L)
+            wav.setsampwidth(4)
             frame = mf.read()
             while frame is not None:
                 wav.writeframes(frame)
