diff -Naur a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2020-11-27 19:10:34.953594132 +1100
+++ b/CMakeLists.txt	2020-11-27 19:15:50.763046476 +1100
@@ -1030,14 +1030,14 @@
     file(WRITE "${pkgconfig_file}" "# draco pkg-config.\n")
     file(APPEND "${pkgconfig_file}" "prefix=${prefix}\n")
     file(APPEND "${pkgconfig_file}" "exec_prefix=${prefix}/bin\n")
-    file(APPEND "${pkgconfig_file}" "libdir=${prefix}/lib\n")
+    file(APPEND "${pkgconfig_file}" "libdir=${prefix}/lib${LIB_SUFFIX}\n")
     file(APPEND "${pkgconfig_file}" "includedir=${prefix}/include\n\n")
     file(APPEND "${pkgconfig_file}" "Name: Draco\n")
     file(APPEND "${pkgconfig_file}" "Description: 3D Compression.\n")
     file(APPEND "${pkgconfig_file}" "Version: ${draco_version}\n")
     file(APPEND "${pkgconfig_file}" "Requires:\n")
     file(APPEND "${pkgconfig_file}" "Conflicts:\n")
-    file(APPEND "${pkgconfig_file}" "Libs: -L${prefix}/lib -ldraco\n")
+    file(APPEND "${pkgconfig_file}" "Libs: -L${prefix}/lib${LIB_SUFFIX} -ldraco\n")
     file(APPEND "${pkgconfig_file}" "Cflags: -I${prefix}/include -std=c++11\n")
   endif()
 
@@ -1069,18 +1069,18 @@
   if(NOT MSVC)
     # Install our pkg-config file.
     install(FILES "${draco_build_dir}/draco.pc"
-            DESTINATION "${CMAKE_INSTALL_PREFIX}/lib/pkgconfig")
+            DESTINATION "${CMAKE_INSTALL_PREFIX}/lib${LIB_SUFFIX}/pkgconfig")
   endif()
 
   # Add install rules for lib and executable targets.
   install(TARGETS dracodec dracoenc draco DESTINATION
-                  "${CMAKE_INSTALL_PREFIX}/lib")
+                  "${CMAKE_INSTALL_PREFIX}/lib${LIB_SUFFIX}")
   if(BUILD_UNITY_PLUGIN)
-    install(TARGETS dracodec_unity DESTINATION "${CMAKE_INSTALL_PREFIX}/lib")
+    install(TARGETS dracodec_unity DESTINATION "${CMAKE_INSTALL_PREFIX}/lib${LIB_SUFFIX}")
   endif()
   if(BUILD_MAYA_PLUGIN)
     install(
-      TARGETS draco_maya_wrapper DESTINATION "${CMAKE_INSTALL_PREFIX}/lib")
+      TARGETS draco_maya_wrapper DESTINATION "${CMAKE_INSTALL_PREFIX}/lib${LIB_SUFFIX}")
   endif()
 
   install(TARGETS draco_decoder draco_encoder DESTINATION
@@ -1088,9 +1088,9 @@
 
   # Generate a CMake config file.
   set(draco_include_install_dir "include/draco")
-  set(draco_lib_install_dir "lib")
+  set(draco_lib_install_dir "lib${LIB_SUFFIX}")
   set(draco_script_install_dir
-      "${CMAKE_INSTALL_PREFIX}/${draco_lib_install_dir}/draco/cmake")
+      "${CMAKE_INSTALL_PREFIX}/${draco_lib_install_dir}/cmake/draco")
   set(draco_config_in_file "${draco_root}/cmake/DracoConfig.cmake")
   set(draco_config_out_file "${draco_build_dir}/DracoConfig.cmake")
 
@@ -1098,7 +1098,7 @@
                                 "${draco_config_out_file}"
                                 NO_CHECK_REQUIRED_COMPONENTS_MACRO
                                 INSTALL_DESTINATION
-                                "${draco_lib_install_dir}/draco/cmake" PATH_VARS
+                                "${draco_lib_install_dir}/cmake/draco" PATH_VARS
                                 draco_include_install_dir draco_lib_install_dir)
   write_basic_package_version_file("${draco_build_dir}/DracoConfigVersion.cmake"
                                    VERSION ${draco_version}
