diff -Naur a/opencv_contrib-3.1.0/modules/text/CMakeLists.txt b/opencv_contrib-3.1.0/modules/text/CMakeLists.txt
--- a/opencv_contrib-3.1.0/modules/text/CMakeLists.txt	2016-06-22 22:30:13.427798189 +0930
+++ b/opencv_contrib-3.1.0/modules/text/CMakeLists.txt	2016-06-22 22:31:41.472796913 +0930
@@ -1,11 +1,13 @@
 set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${CMAKE_CURRENT_SOURCE_DIR})
 
-find_package(Tesseract)
-if(Tesseract_FOUND)
-    message(STATUS "Tesseract:   YES")
-    set(HAVE_TESSERACT 1)
-else()
-    message(STATUS "Tesseract:   NO")
+if (WITH_TESSERACT)
+    find_package(Tesseract)
+    if(Tesseract_FOUND)
+        message(STATUS "Tesseract:   YES")
+        set(HAVE_TESSERACT 1)
+    else()
+        message(STATUS "Tesseract:   NO")
+    endif()
 endif()
 
 configure_file(${CMAKE_CURRENT_SOURCE_DIR}/text_config.hpp.in
