diff -Naur a/OgreMeshy/CMakeLists.txt b/OgreMeshy/CMakeLists.txt
--- a/CMakeLists.txt	2013-05-23 13:47:25.000000000 +1000
+++ b/CMakeLists.txt	2017-05-15 02:24:03.741081627 +1000
@@ -102,6 +102,7 @@
 if(UNIX)
 	TARGET_LINK_LIBRARIES( OgreMeshy ${GTK2_LIBRARIES} )
 	TARGET_LINK_LIBRARIES( OgreMeshy ${OPENGL_LIBRARIES} )
+	TARGET_LINK_LIBRARIES( OgreMeshy /usr/lib/libX11.so )
 endif(UNIX)
 
 set(INST_DEST ".")
@@ -188,4 +189,4 @@
 	SET(CPACK_SOURCE_STRIP_FILES "")
 ENDIF(WIN32 AND NOT UNIX)
 	#SET(CPACK_PACKAGE_EXECUTABLES "MyExecutable" "My Executable")
-INCLUDE(CPack)
\ No newline at end of file
+INCLUDE(CPack)
